test/http/serve-delay-connect.go

```diff
@@ -0,0 +1,119 @@
+package http
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/florianl/go-nfqueue"
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultPort     = "8080"
+	defaultDelay    = time.Duration(40 * time.Second)
+	defaultQueueNum = 100
+)
+
+func NewServeDelayConnectCommand() *cobra.Command {
+	var command = &cobra.Command{
+		Use:   "serve-delay-connect-test-server",
+		Short: "Runs test http server which delays the connection accept and sends echo response.",
+		Long:  "Runs test http server which delays the connection accept and sends echo response.",
+		Run: func(cmd *cobra.Command, args []string) {
+			queueNum, err := strconv.ParseUint(os.Getenv("QUEUE"), 10, 16)
+			if err != nil {
+				queueNum = defaultQueueNum
+			}
+			delay, err := time.ParseDuration(os.Getenv("DELAY"))
+			if err != nil {
+				delay = defaultDelay
+			}
+			port := os.Getenv("PORT")
+			if len(port) == 0 {
+				port = defaultPort
+			}
+			if err := serveDelayConnect(uint16(queueNum), delay, port); err != nil {
+				log.Printf("Error received: %v\n", err)
+			}
+		},
+	}
+
+	return command
+}
+
+func serveDelayConnect(queueNum uint16, delay time.Duration, port string) error {
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Hmm, I can't actually ctrl-c out of this program when I run it on the command line:
```
sudo ./ingress-operator serve-delay-connect-test-server
[sudo] password for gspence: 
2024/06/12 16:32:46 Callback is registered with delay 40s
2024/06/12 16:32:46 Serving on 8080
^[[1;5A^C2024/06/12 17:34:30 Received signal: interrupt
2024/06/12 17:34:30 Error reading from netlink socket: netlink receive: i/o timeout
^C^C^C^C
^C^C^C^C
```

I think it's not actually stopping the program. Does this do the same for you?

Reply:
Good catch! The problem was in the unnecessary signal handling, removed it.